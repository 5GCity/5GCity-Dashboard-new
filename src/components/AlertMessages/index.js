/**
 * AlertMessages Component
 * Please write a description
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import styled from 'styled-components'
import { Icon } from 'element-react'

export default ({ children, ...props }) => (
  <Wrapper {...props}>
    {/* <Close onClick={() => props.close()}>&times;</Close> */}
    <Circle>
      <IconRemove name={'close'} />
    </Circle>
    <Container>
      <Title>{props.title}</Title>
    </Container>
      {props.location &&
      <Container>
        <SubTitle>Location:</SubTitle>
        <Location>{props.location}</Location>
      </Container>
      }
      <Container>
        <SubTitle>Description:</SubTitle>
        <Description>{props.description}</Description>
      </Container>
  </Wrapper>
)

const Wrapper = styled.div`
  height: 108px;
  width: 320px;
  opacity: 0.98;
  border-radius: 6px;
  background-color: #EFF2F7;
  box-shadow: 0 0 20px 0 rgba(0,0,0,0.15);
  margin: 10px auto;
`
const Circle = styled.div`
  height:48px;
  width: 48px;
  background-color:#D84F4F;
  border-radius: 50%;
  position: relative;
  top: 24px;
  left: -24px;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2), 0 0 10px 0 rgba(216,79,79,0.5);
`
const Title = styled.p`
  margin: 4px 0;
  font-size: 14px;
  font-family: ${({ theme }) => theme.fontDin };
  letter-spacing: 0.5px;
  line-height: 20px;
  color: #D84F4F;
  font-weight: bold;
`
const Location = styled.span`
  font-size: 12px;
  font-weight: 600;
  width: 184px;
  line-height: 16px;
  font-family: ${({ theme }) => theme.fontFamily };
`
const Description = styled.div`
  display: inline-flex;
  font-size: 12px;
  font-weight: 600;
  height: 32px;
  width: 184px;
  line-height: 16px;
  font-family: ${({ theme }) => theme.fontFamily };
`
/* const Close = styled.div`
  position: relative;
  top: 0;
  z-index: 2;
  text-align: center;
  float: right;
  height: 32px;
  font-size: 20px;
  width: 32px;
  color: #89979F;
  cursor: pointer;
` */

const Container = styled.div`
  text-align: left;
  position: relative;
  top: -40px;
  margin-left: 40px;
`
const SubTitle = styled.p`
  font-family: ${({ theme }) => theme.fontFamily };
  font-weight: bold;
  font-size: 12px;
  line-height: 16px;
  display: inline-block;
  color: #37474F;
  width: 80px;
  margin: 4px 0;
`
const IconRemove = styled(Icon)`
  color:#fff;
  position: relative;
  top: 13px;
  left: 15px;
`
