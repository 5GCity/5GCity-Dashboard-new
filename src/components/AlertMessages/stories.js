/**
 * AlertMessages Component Stories
 * Please write a description or remove this line
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import { storiesOf } from '@storybook/react'

import AlertMessages from './index'
const alerts = [
  {id: 1, type: 'danger', title:'asdkasjdkj-asd', location: 'eu.5gcity-nfv.vnf.0.3', description: 'App has unused connection points: [vdu02:eth3]'},
  {id: 2, type: 'danger', title:'evt_vnfd_itg_unused_cpoint', location: 'eu.5gcity-nfv.vnf.0.3', description: 'connection points'}
]
const State1 = () => (
  <div>
  <AlertMessages
    title={alerts[0].title}
    location={alerts[0].location}
    description={alerts[0].description}
    type={alerts[0].type}
  />
    <AlertMessages
    title={alerts[1].title}
    location={alerts[1].location}
    description={alerts[1].description}
    type={alerts[1].type}
  />
  </div>
)

storiesOf('AlertMessages', module)
  .add('primary', State1)
