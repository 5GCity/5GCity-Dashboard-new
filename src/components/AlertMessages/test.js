/**
 * AlertMessages Component Tests
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import renderer from 'react-test-renderer'
import AlertMessages from './index'
import 'jest-styled-components'

it('renders correctly', () => {
  const tree = renderer
    .create(<AlertMessages />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
