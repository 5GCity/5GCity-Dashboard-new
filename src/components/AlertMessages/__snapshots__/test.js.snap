// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  height: 108px;
  width: 320px;
  opacity: 0.98;
  border-radius: 6px;
  background-color: #EFF2F7;
  box-shadow: 0 0 20px 0 rgba(0,0,0,0.15);
  margin: 10px auto;
}

.c1 {
  height: 48px;
  width: 48px;
  background-color: #D84F4F;
  border-radius: 50%;
  position: relative;
  top: 24px;
  left: -24px;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2),0 0 10px 0 rgba(216,79,79,0.5);
}

.c2 {
  color: #fff;
  position: relative;
  top: 13px;
  left: 15px;
}

.c3 {
  text-align: left;
  position: relative;
  top: -40px;
  margin-left: 40px;
}

.c4 {
  margin: 4px 0;
  font-size: 14px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  line-height: 20px;
  color: #D84F4F;
  font-weight: bold;
}

.c5 {
  font-weight: bold;
  font-size: 12px;
  line-height: 16px;
  display: inline-block;
  color: #37474F;
  width: 80px;
  margin: 4px 0;
}

.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 12px;
  font-weight: 600;
  height: 32px;
  width: 184px;
  line-height: 16px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <i
      className="el-icon-close c2"
      style={Object {}}
    />
  </div>
  <div
    className="c3"
  >
    <p
      className="c4"
    />
  </div>
  <div
    className="c3"
  >
    <p
      className="c5"
    >
      Description:
    </p>
    <div
      className="c6"
    />
  </div>
</div>
`;
